agent/src/beerocks/fronthaul_manager/fronthaul_manager_main.cpp

```diff
@@ -147,60 +178,82 @@ int main(int argc, char *argv[])
 
-    // Redirect stdout / stderr to file
-    if (logger.get_log_files_enabled()) {
-        beerocks::os_utils::redirect_console_std(beerocks_slave_conf.sLog.files_path +
-                                                 base_monitor_name + "_std.log");
+    // Init logger ap_manager
+    std::string base_ap_manager_name =
+        std::string(syslog ? BEEROCKS_FRONTHAUL : BEEROCKS_AP_MANAGER) + "_" + fronthaul_iface;
+    g_logger_ap_mananger = init_logger(base_ap_manager_name, beerocks_slave_conf.sLog, argc, argv);
+    if (!g_logger_ap_mananger) {
+        return 0;
+    }
+
+    // Init logger monitor
+    std::string base_monitor_name =
+        std::string(syslog ? BEEROCKS_FRONTHAUL : BEEROCKS_MONITOR) + "_" + fronthaul_iface;
+    g_logger_monitor =
+        init_logger(base_monitor_name, beerocks_slave_conf.sLog, argc, argv, BEEROCKS_MONITOR);
+    if (!g_logger_monitor) {
+        return 0;
     }
 
-    //kill running monitor and write pid file
-    beerocks::os_utils::kill_pid(beerocks_slave_conf.temp_path, base_monitor_name);
-    beerocks::os_utils::write_pid_file(beerocks_slave_conf.temp_path, base_monitor_name);
+    // Kill running fronthaul and write pid file
+    std::string base_fronthaul_name = std::string(BEEROCKS_FRONTHAUL) + "_" + fronthaul_iface;
+    beerocks::os_utils::kill_pid(beerocks_slave_conf.temp_path, base_fronthaul_name);
+    beerocks::os_utils::write_pid_file(beerocks_slave_conf.temp_path, base_fronthaul_name);
     std::string pid_file_path =
-        beerocks_slave_conf.temp_path + "pid/" + base_monitor_name; // for file touching
-
-    // start monitor
-    std::string slave_uds =
-        beerocks_slave_conf.temp_path + std::string(BEEROCKS_SLAVE_UDS) + "_" + monitor_iface;
-    son::monitor_thread monitor(slave_uds, monitor_iface, beerocks_slave_conf, logger);
-    if (monitor.init()) {
-        auto touch_time_stamp_timeout = std::chrono::steady_clock::now();
-        while (g_running) {
-
-            // Handle signals
-            if (s_signal) {
-                handle_signal();
-                continue;
-            }
+        beerocks_slave_conf.temp_path + "pid/" + base_fronthaul_name; // For file touching
 
-            if (std::chrono::steady_clock::now() > touch_time_stamp_timeout) {
-                beerocks::os_utils::touch_pid_file(pid_file_path);
-                touch_time_stamp_timeout =
-                    std::chrono::steady_clock::now() +
-                    std::chrono::seconds(beerocks::TOUCH_PID_TIMEOUT_SECONDS);
-            }
+    // Get Agent UDS file
+    std::string agent_uds =
+        beerocks_slave_conf.temp_path + std::string(BEEROCKS_SLAVE_UDS) + "_" + fronthaul_iface;
+
+    // Create ap_manager
+    son::ap_manager_thread ap_manager(agent_uds, fronthaul_iface, *g_logger_ap_mananger);
+
+    if (!ap_manager.init()) {
+        CLOG(ERROR, g_logger_ap_mananger->get_logger_id()) << "ap manager init() has failed!";
+        return 0;
```

Comment:
That's an error, so you should probably return a non-zero value

Reply:
Done, but changed only this return value.
This PR is big enough and I will need to do a separate commit for this.
So probably it will be better to change it here and on any other process as part of a different PR.